### DISCUSSION TITLE
External site links not working

### DISCUSSION THREAD

**Discussion by mmpadilla15** [thegetty/quire#535] [Get Help] [answered]
All of my links work just fine in preview, but when I deploy to Netlify, we get a 404 error:

![Screen Shot 2022-10-10 at 2 01 50 PM](https://user-images.githubusercontent.com/74433007/194937126-407ff8c1-d4fd-4a7f-bd58-2c5eae36e739.png)

Netlify's Support Guide talks about needing a /index file, but these are external links, so it doesn't seem like they should need a /index file.

Can someone advise?

**Comment by geealbers** [comment, marked as answer]
I think we'll need more information in order to help troubleshoot. Can you please provide a link to your site on Netlify and/or information on specific links that are having this issue?

**Reply by mmpadilla15** [reply]
@geealbers Sure! Here's the link: [https://art-scene.cartermuseum.org/](https://art-scene.cartermuseum.org/)

Initially, I thought it was just the About and Privacy Policy links at the bottom of the sidebar TOC. But I started clicking around and these are the others I've found:

- Plate 7 View in Collection link
- Plate 8 View in Collection link
- Plate 29 View in Collection link
- Contributors page: Links next to Edwards and Frembling

As I'm pulling these up now, I'm noticing something else ... all the links that get the 404 errors are external links that are opening in a new window. All of our other links that go to external sources open in the same window and are working, except that back in August I thought I had it set up to open all external links in new windows. See #452.

Hmmm ...

**Reply by geealbers** [reply]
Thanks for the additional information. First, if you have a `config/_default/` directory in your project, delete it and all its contents. And then try adding `absURLs: true` to your `config/site.yml` file and redeploying. Like ...

```
baseURL: "/"
relativeURLs: true
absURLs: true
```

**Reply by mmpadilla15** [reply]
@geealbers OK, this _sort of_ works. The links that weren't working at all are now working - yay! But links in the footnotes throughout and in the Exhibitions section of the Plates are all opening in the same window instead of a new window.

Thoughts?

**Reply by geealbers** [reply]
I'm not sure. To start, can you confirm that in your project you have a `render-link.html` file at either `layouts/_default/_markup/render-link.html` OR `themes/default/layouts/_default/_markup/render-link.html` and the contents of that file is this?:

```html
<a href="{{ .Destination | safeURL }}"{{ with .Title}} title="{{ . }}"{{ end }}{{ if strings.HasPrefix .Destination "http" }} target="_blank" rel="noopener"{{ end }}>{{ .Text | safeHTML }}</a>
```

**Reply by mmpadilla15** [reply]
@geealbers I checked my `themes/default/layouts/_default/` and there wasn't a `_markup` folder there. But I found the folder with the `render-link.html` file in it in a previous version that I had been working on and moved it to that folder. And yes, the contents of the `render-link.html` file matches what you have above. So I saved and redeployed.

Now all external links are opening in a new window EXCEPT for the links in the Exhibitions sections of the Plates:
![Screen Shot 2022-10-14 at 2 59 07 PM](https://user-images.githubusercontent.com/74433007/195932776-cbaa1305-21f5-4bff-8a9e-7e73701b6aa4.png)
For some reason, those continue to open in the same window. Those links are part of the `{{< q-class "backmatter" >}}` section on the plates. Could that have something to do with it?

**Reply by geealbers** [reply]
That's odd. I wouldn't have thought the `q-class` shortcode would have been an issue in this case, but I also don't have any other ideas. You could replace that shortcode with HTML markup and see if it makes a difference.

```
{{< q-class "backmatter" >}}
...
{{< /q-class >}}
```
Can be written in HTML as

```html
<div class="backmatter">
...
</div>
```

**Reply by mmpadilla15** [reply]
@geealbers I just tested with the `<div>` code on one Plate and deployed and the external link is now opening in a new window! It works - yay! Thank you so much!
